docs/sources/reference/components/loki/loki.source.file.md

```diff
@@ -105,6 +109,27 @@ If no file changes are detected, the poll frequency doubles until a file change
 
 If file changes are detected, the poll frequency is reset to `min_poll_frequency`.
 
+### `file_match`
+
+The `file_match` block enables built-in file discovery directly within `loki.source.file`.
+When enabled, the component discovers files on the local filesystem using glob patterns and the [doublestar][] library.
+
+The following arguments are supported:
+
+| Name                | Type                | Description                                                 | Default | Required |
+|---------------------|---------------------|-------------------------------------------------------------|---------|----------|
+| `enabled`           | `bool`              | Enable file discovery using glob patterns.                  | `false` | no       |
+| `ignore_older_than` | `duration`          | Ignores files with modification times before this duration. | `"0s"`  | no       |
+| `sync_period`       | `duration`          | How often to sync filesystem and targets.                   | `"10s"` | no       |
+
+Benefits of using `file_match` over `local.file_match`:
+- Simplified configuration: No need for a separate `local.file_match` component
+- Better performance: File discovery is integrated directly into `loki.source.file`, eliminating the overhead of component-to-component communication
+- Same functionality: Supports the same glob patterns and features, including `__path_exclude__` for excluding files
+
+When `enabled` is set to `true`, you can use glob patterns (e.g., `/tmp/*.log` or `/var/log/**/*.log`) directly in the `targets` argument's `__path__` label.
+The component periodically scans the filesystem based on `sync_period` and automatically discovers new files, removes deleted files, and ignores files older than `ignore_older_than` if specified.
```

Comment:
Didn't we have fsnotify-based implementation where it's supported?

Reply:
I don't think so. The only usage I can see of fsnotify is https://github.com/grafana/alloy/blob/main/internal/filedetector/detector.go#L68 and this is used by `local.file` and not `local.file_match`.